comm.go

```diff
@@ -54,10 +65,17 @@ func (p *PubSub) handleNewStream(s network.Stream) {
 				// but it doesn't hurt to send it.
 				s.Close()
 			}
+
+			p.inboundStreamsMx.Lock()
+			if p.inboundStreams[peer] == s {
+				delete(p.inboundStreams, peer)
+			}
+			p.inboundStreamsMx.Unlock()
```

Comment:
Probably should put this in a `defer`, so that it's also run on other return paths.

Reply:
Well, there are no other relevant return code paths. The only other return is in the context cancellation case, where we don't care because we are shutting down.
Having said that, it is probably cleaner to do it in a defer so why not.